analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -88,6 +90,15 @@ open class Npm(
 
     private val ortProxySelector = installAuthenticatorAndProxySelector()
 
+    private val npmRcFile = Os.userHomeDirectory.resolve(".npmrc")
+
+    private val npmRegistry: String =
+        if (npmRcFile.isFile) {
+            readRegistryFromNpmRc(npmRcFile.readText()) ?: PUBLIC_NPM_REGISTRY
```

Comment:
Actually, I liked you previous code (and my suggestion to use an `init` block) better as it read the `.npmrc` file only once. Any reason not to follow the `init` block approach here?

Reply:
I follow Intellij advice ;-) 
Ok to use `init` block

Reply:
.npmrc is also used in `fun beforeResolution(definitionFiles: List<File>)`
so I'll keep `private val npmRcFile: File` and initialize it in the `init` block

Reply:
Please take a look again at [my proposal](https://github.com/oss-review-toolkit/ort/pull/3669#discussion_r581094004), it also moves

    ortProxySelector.addProxies(managerName, readProxySettingsFromNpmRc(npmRcContent))

from `beforeResolution()` to the `init` block.

Reply:
Hum... Ok !  I thought it was an oversight due to a copy/paste of my initial code. ;-)
I'll change that.
So `beforeResolution` will only have the checkVersion

Reply:
> So `beforeResolution` will only have the checkVersion

Exactly!

Reply:
However, I believe it makes more sense to squash the next commit to this one. Otherwise, when just looking at this commit, one might be wondering why you move original line 109 to lines 94 / 99, but *not* move original lines 110 - 112 also, just in order to change this in the next commit. I'd argue in the commit message with an additional paragraph:

> While at it, also move the proxy handling from `beforeResolution()` to the `init` block in order to read the `.npmrc` file only once, and as it is probably safe to assume that the `.npmrc` file does not change between resolution runs.